solr/core/src/test/org/apache/solr/cloud/TestRebalanceLeaders.java

```diff
@@ -604,74 +572,61 @@ private void forceUpdateCollectionStatus() {
 
   // Since we have to restart jettys, we don't want to try re-balancing etc. until we're sure all
   // jettys that should be up are and all replicas are active.
-  private void checkReplicasInactive(List<JettySolrRunner> downJettys) throws InterruptedException {
-    TimeOut timeout = new TimeOut(timeoutMs, TimeUnit.MILLISECONDS, TimeSource.NANO_TIME);
-    DocCollection docCollection = null;
-    Set<String> liveNodes = null;
+  private void checkReplicasInactive(List<JettySolrRunner> downJettys) {
 
     Set<String> downJettyNodes = new TreeSet<>();
     for (JettySolrRunner jetty : downJettys) {
       downJettyNodes.add(
           jetty.getBaseUrl().getHost() + ":" + jetty.getBaseUrl().getPort() + "_solr");
     }
-    while (timeout.hasTimedOut() == false) {
-      forceUpdateCollectionStatus();
-      docCollection = cluster.getSolrClient().getClusterState().getCollection(COLLECTION_NAME);
-      liveNodes = cluster.getSolrClient().getClusterState().getLiveNodes();
-      boolean expectedInactive = true;
-
-      for (Slice slice : docCollection.getSlices()) {
-        for (Replica rep : slice.getReplicas()) {
-          if (downJettyNodes.contains(rep.getNodeName()) == false) {
-            continue; // We are on a live node
-          }
-          // A replica on an allegedly down node is reported as active.
-          if (rep.isActive(liveNodes)) {
-            expectedInactive = false;
+
+    waitForState(
+        "Waiting for all replicas to become inactive",
+        COLLECTION_NAME,
+        (liveNodes, docCollection) -> {
+          boolean expectedInactive = true;
+
+          for (Slice slice : docCollection.getSlices()) {
+            for (Replica rep : slice.getReplicas()) {
+              if (!downJettyNodes.contains(rep.getNodeName())) {
```

Comment:
I don't recommend flipping this wherever you see it, but you are quite welcome to use either style.  Over 10 years ago, Lucene (which included Solr) developers started preferring the higher clarity of == false over an exclamation mark, I recall related to the visual subtleness of a single char.  Nevertheless there's no code standard on this.

Reply:
We really should establish a code standard and have it be enforced....

Reply:
There doesn't have to be a code standard for everything Eric :-)

Reply:
;-).  I think we will have to agree to disagree ;-).   I really rely on the tools to encourage me in the right way to do things, or we end up with multiple ways.  I also have the memory of a gold fish, so if it isn't enforced by a tool, I may say "Hey, it should be X" for a year or two, and then swap to "Hey, it should be Y" just because I can't remember what the agreement was.   I can imagine that too strict standards becomes a strait jacket too.

Reply:
I flipped this because it is reported by my IDE. I didn't know such a convention was embraced at some point. Will change back to `== false` form.